### DISCUSSION TITLE
Restore backup does not update the app

### DISCUSSION THREAD

**Discussion by ginex25** [simolus3/drift#2216] [Q&A]
Hi! 

I have in my app the possibility to export and import the database. 
Importing generally works, but unfortunately you have to restart the app to see the data. 
Is there any way to display the data without restarting the app?

I use sqlite3 to open the backup database and the custom statement 'VACCUM INTO ?' to load the data.

**Comment by simolus3** [comment]
> I use sqlite3 to open the backup database and the custom statement 'VACCUM INTO ?' to load the data.

Can you describe that a bit more? Are you using the backup APIs to create the backup files?

When using `VACUUM INTO` a file that's already opened, I think that data should be visible as well. Do you just not see it because query streams don't update or do other queries/new streams not work either? If it's just about query streams, you could use something like this to update them manually:

```dart
extension on GeneratedDatabase {
  void reload() {
    markTablesUpdated(allTables);
  }
}
```

**Reply by ginex25** [reply]
> Can you describe that a bit more? 

My code for importing looks like this: 

    final backupDB= sqlite3.open(path);
    final appDir = await getApplicationDocumentsDirectory();
    final File file = File(p.join(appDir.path, 'meter.db'));

    backupDB.execute('VACUUM INTO ?', [file.path]);

    backupDB.dispose();


> Are you using the backup APIs to create the backup files?

I create a backup exactly as it is given on your example page.  I just use another package to determine the path.



After I tried to use the markTablesUpdated method, which unfortunately didn't bring any change, I simply tried to create a new entry. This should be possible if the database is already open, but I got the error:  Unhandled Exception: Unable to load asset: "path"
Do I have to reload the database once and if so, how do I do that during runtime?
